frontend/public/components/overview.jsx

```diff
@@ -0,0 +1,353 @@
+import { FLAGS, connectToFlags, flagPending } from '../features';
+import { Dropdown, Firehose, NavTitle, StatusBox } from './utils';
+import { TextFilter } from './factory';
+import { ProjectOverview } from './project-overview';
+
+const getOwnedResources = (resources, uid) => {
+  return _.filter(resources, function (resource) {
+    const {ownerReferences} = resource.metadata;
+    return _.some(ownerReferences, {
+      uid,
+      controller: true
+    });
+  });
+};
+
+const OverviewToolbar = ({groupOptions, handleFilterChange, handleGroupChange, selectedGroup}) =>
+  <Toolbar className="overview-toolbar">
+    <Toolbar.RightContent>
+      <div className="form-group overview-toolbar__form-group">
+        <label className="overview-toolbar__label">
+          Group by label
+        </label>
+        <Dropdown
+          className="overview-toolbar__dropdown"
+          disabled={_.isEmpty(groupOptions)}
+          items={groupOptions}
+          onChange={handleGroupChange}
+          style={{display: 'inline-block'}}
+          title={selectedGroup}
+        />
+      </div>
+      <div className="form-group overview-toolbar__form-group">
+        <TextFilter
+          label="Resources by name"
+          onChange={handleFilterChange}
+        />
+      </div>
+    </Toolbar.RightContent>
+  </Toolbar>;
+
+OverviewToolbar.displayName = 'OverviewToolbar';
+
+OverviewToolbar.propTypes = {
+  groupOptions: PropTypes.object.isRequired,
+  handleFilterChange: PropTypes.func.isRequired,
+  handleGroupChange: PropTypes.func.isRequired,
+  selectedGroup: PropTypes.string.isRequired
+};
+
+class OverviewDetails extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleFilterChange = this.handleFilterChange.bind(this);
+    this.handleGroupChange = this.handleGroupChange.bind(this);
+    this.clearFilter = this.clearFilter.bind(this);
+
+    this.state = {
+      filterValue: '',
+      items: [],
+      filteredItems: [],
+      groupedItems: [],
+      groupOptions: {},
+      selectedGroupLabel: 'app'
+    };
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (this.props.namespace !== prevProps.namespace
+      || this.props.replicationControllers !== prevProps.replicationControllers
+      || this.props.pods !== prevProps.pods
+      || this.props.deploymentConfigs !== prevProps.deploymentConfigs) {
+      this.createOverviewData();
+    } else if (this.state.filterValue !== prevState.filterValue) {
+      const filteredItems = this.filterItems(this.state.items);
+      this.setState({
+        filteredItems,
+        groupedItems: this.groupByLabel(filteredItems, this.state.selectedGroupLabel)
+      });
+    } else if (this.state.selectedGroupLabel !== prevState.selectedGroupLabel) {
+      this.setState({
+        groupedItems: this.groupByLabel(this.state.filteredItems, this.state.selectedGroupLabel)
+      });
+    }
+  }
+
+  sortByDeploymentVersion(replicationControllers, descending) {
```

Comment:
It's not clear to me why we need to sort by deployment version if we're only showing the active one. We might be able to skip this for now?

Reply:
My thought was that we will have to use some logic similar to this in order to find the replication controller with the latest version. Since we will eventually be showing the whole list, sorting and grabbing the head seems like a good way to do it to make things easier for showing the whole list. We could make this into something like 'getLatestDeploymentVersion' but we would still have to iterate over all the RC's and do this same logic to return one RC.

Reply:
You should be able to look at the property on the deployment config or deployment to get the current revision. rc-10 might be failed and rc-9 is what is running... It's not necessarily the most recent by number

Reply:
Gotcha, okay, that makes sense